Fix error detection

lukasmerk opened the pull request:
Currently, the code doesn't contain error messages from the Azure backend for me.

The previous error detection code seems to have been copied from stevenmaguire/oauth2-microsoft, but the response format for errors seem to be different when using Azure. The JSON in the body looks like this for me:
```json
{
    "error": "invalid_client",
    "error_description": "AADSTS50011: The redirect URI … [redacted for privacy]",
    "error_codes": [
        50011
    ],
    "timestamp": "2023-12-11 11:06:35Z",
    "trace_id": "[redacted for privacy]",
    "correlation_id": "[redacted for privacy]",
    "error_uri": "https://login.microsoftonline.com/error?code=50011"
}
```

I changed the code which creates the Exception accordingly, and this works for me. The Exception previously only had a generic "Bad Request" error message, with the change it is the one from Azure.

lukasmerk committed:
Fix error detection

The previous error detection code seems to have been copied from stevenmaguire/oauth2-microsoft, but the response format for errors seem to be different when using Azure. 
Part of an example JSON:
{
    "error": "invalid_client",
    "error_description": "[…]",
…

greew commented:
Great catch - the tests fails thogh. 

In `tests/src/Provider/AzureTest.php` can you change line 162-164 from

```php
$postResponse->shouldReceive('getBody')->andReturn(
    '{"error": {"code": "request_token_expired", "message": "' . $message . '"}}'
);
```

to 

```php
$postResponse->shouldReceive('getBody')->andReturn(
    '{"error": "request_token_expired", "error_description": "' . $message . '"}'
);
```

lukasmerk committed:
Update Test to match new Error Detection

Aligns the test with the changes from 8bb1c6d69cb754467b0d92a40f2ca5019ab2ce2d, see PR #1

lukasmerk commented:
Thanks for your feedback! I updated the tests accordingly.
